src/state.rs

```diff
@@ -101,3 +146,20 @@ pub async fn epoch_loop(state: OPRFState, config: &Config) {
         info!("epoch now {}", s.epoch);
     }
 }
+
+/// Parse a timestamp out of the Config
+impl From<&Config> for Option<time::OffsetDateTime> {
+    fn from(config: &Config) -> Self {
+        let mut basetime = None;
+        if let Some(stamp) = &config.epoch_basetime {
+            basetime = match time::OffsetDateTime::parse(stamp, &Rfc3339) {
+                Ok(timestamp) => Some(timestamp),
+                Err(e) => {
+                    warn!("Couldn't parse epoch-basetime argument: {e}");
```

Comment:
does it make sense to silently fail here? this seems like a potentially critical error, and we may want to quit the application if the user-defined value is not valid.

also, i'm unsure if implementing `From<&Config>` for parsing a single field is good for extensibility. if we wanted to add multiple timestamps to the config, then we will have to gut this trait impl.

we could have a func called `parse_timestamp` that returns the parsing result directly. in the config struct, we should be able to use the type and the function directly:

```
#[arg(value_parser = parse_timestamp)]
epoch_basetime: Option<time::OffsetDateTime>
```

Reply:
here is some more info regarding custom parsers in clap: https://docs.rs/clap/latest/clap/_derive/_tutorial/index.html#validated-values

Reply:
Yes, the `From<&Config>` is too cute. I'll make it a function and see if we can parse directly to avoid all the option-flattening gymnastics.